modules/common.nix

```diff
@@ -0,0 +1,160 @@
+# The common module that contains Hjem's per-user options. To ensure Hjem remains
+# somewhat compliant with cross-platform paradigms (e.g. NixOS or Darwin) distro
+# specific options, such as packages, must be avoided here.
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (lib.modules) mkIf mkDefault mkDerivedConfig;
+  inherit (lib.options) mkOption mkEnableOption literalExpression;
+  inherit (lib.strings) hasPrefix;
+  inherit (lib.types) bool submodule str path attrsOf nullOr lines listOf package;
+
+  clobberOpt = config.clobberFiles;
+
+  fileType = relativeTo:
+    submodule ({
+      name,
+      target,
+      config,
+      options,
+      ...
+    }: {
+      options = {
+        enable = mkOption {
+          type = bool;
+          default = true;
+          example = false;
+          description = ''
+            Whether this file should be generated. If set to `false`, the file will
+            not be created.
+          '';
+        };
+
+        target = mkOption {
+          type = str;
+          apply = p:
+            if hasPrefix "/" p
+            then throw "This option cannot handle absolute paths yet!"
+            else "${config.relativeTo}/${p}";
+          defaultText = "name";
+          description = ''
+            Path to target file relative to ${config.relativeTo}.
+          '';
+        };
+
+        text = mkOption {
+          default = null;
+          type = nullOr lines;
+          description = "Text of the file";
+        };
+
+        source = mkOption {
+          type = nullOr path;
+          default = null;
+          description = "Path of the source file or directory";
+        };
+
+        executable = mkOption {
+          type = bool;
+          default = false;
+          example = true;
+          description = ''
+            Whether to set the execute bit on the target file.
+          '';
+        };
+
+        clobber = mkOption {
+          type = bool;
+          default = clobberOpt;
+          defaultText = literalExpression "config.hjem.clobberByDefault";
+          description = ''
+            Whether to "clobber" existing target paths. While `true`, tmpfile rules will be constructed
```

Comment:
mentioning `systemd-tmpfiles` seems linux-specific, maybe drop second sentence?

Reply:
Sounds about right, systemd-tmpfiles won't be the only way to manage links in the future.